lib/yggdrasil/player.ex

```diff
@@ -12,9 +13,8 @@ defmodule Yggdrasil.Player do
     Supervisor.add_player user_id, [owner_pid, push_msg]
   end
 
-  def run_cmd(user_id, cmd = %Message{type: :command}) do
-    player_pid = Registry.get_player user_id
-    GenServer.cast(player_pid, {:run_cmd, cmd})
+  def run_cmd(user_id, cmd) do
+    Command.execute cmd, user_id
```

Comment:
That said, I think both agree that the room should handle the actual "execution" of the events and do agree that it would naturally serialize the actions.

All I was suggesting is that commands build the context and we dispatch the context object to the room and as it works through the actions it creates all the side effects etc.